### PR TITLE
[Next] Backport changes from master

### PR DESCRIPTION
**Pull request by slax57**
Backporting changes from 4.16.12 to 4.16.15 to the `next` branch.

### CODE REVIEW

File: docs/useGetList.md [hunk 1]
```diff
@@ -88,6 +90,50 @@ const LatestNews = () => {
...
+    ListContextProvider,
+    Datagrid,
+    TextField,
+    DateField,
+    NumberField,
+    Pagination
+} from 'react-admin';
+
+const LatestNews = () => {
+    const { data, isLoading, error } = useGetList(
```
Comment (fzaninotto) [inline docs/useGetList.md:110]:
```suggestion
    const { data, isPending, error } = useGetList(
```

File: docs/useGetList.md [hunk 2]
```diff
@@ -88,6 +90,50 @@ const LatestNews = () => {
...
+        'posts',
+        { pagination: { page: 1, perPage: 100 } },
+    );
+    if (error) { return <p>ERROR</p>; }
+    const listContext = useList({ 
+        data,
+        isLoading,
```
Comment (fzaninotto) [inline docs/useGetList.md:117]:
```suggestion
        isPending,
```

File: docs/useGetList.md [hunk 3]
```diff
@@ -127,6 +173,52 @@ const LatestNews = () => {
...
+```
+
+is the equivalent of:
+
+```jsx
+import { useGetList } from 'react-admin';
+
+const PostComments = () => {
+    const record = useRecordContext();
+    const { data, isLoading, error } = useGetList(
```
Comment (fzaninotto) [inline docs/useGetList.md:203]:
```suggestion
    const { data, isPending, error } = useGetList(
```

File: docs/useGetList.md [hunk 4]
```diff
@@ -127,6 +173,52 @@ const LatestNews = () => {
...
+        'comments',
+        { filter: { post_id: record.id } }
+    );
+    if (isLoading) { return <Loading />; }
```
Comment (fzaninotto) [inline docs/useGetList.md:207]:
```suggestion
    if (isPending) { return <Loading />; }
```

**Review by fzaninotto — CHANGES_REQUESTED** [review]

File: docs/useGetList.md (hunk 1 above)
Comment (slax57) [reply docs/useGetList.md]: Nice catch! Thanks!

**Review by fzaninotto — APPROVED** [review]

**Pull request merged**